frontend/public/components/namespace.jsx

```diff
@@ -274,53 +273,42 @@ export const ProjectsTable = (props) => (
   />
 );
 
-const ProjectList_ = (props) => {
-  const ProjectEmptyMessageDetail = (
-    <>
-      <p className="co-pre-line">
-        {props.createProjectMessage || 'Create a project for your application.'}
-      </p>
-      <p>
-        To learn more, visit the OpenShift{' '}
-        <ExternalLink href={openshiftHelpBase} text="documentation" />.
-      </p>
-      <p>
-        Download the <Link to="/command-line-tools">command-line tools</Link>
-      </p>
-    </>
-  );
+const ProjectList = ({ canCreate, data, ...rest }) => {
   const ProjectEmptyMessage = () => (
-    <MsgBox title="Welcome to OpenShift" detail={ProjectEmptyMessageDetail} />
+    <MsgBox
+      title="Welcome to OpenShift"
+      detail={<OpenShiftGettingStarted canCreateProject={canCreate} />}
+    />
   );
   const ProjectNotFoundMessage = () => <MsgBox title="No Projects Found" />;
   return (
     <Table
-      {...props}
+      {...rest}
       aria-label="Projects"
+      data={data}
       Header={ProjectTableHeader}
       Row={ProjectTableRow}
-      EmptyMsg={props.data.length > 0 ? ProjectNotFoundMessage : ProjectEmptyMessage}
+      EmptyMsg={data.length > 0 ? ProjectNotFoundMessage : ProjectEmptyMessage}
       virtualize
     />
   );
 };
-export const ProjectList = connect(createProjectMessageStateToProps)(ProjectList_);
 
-const ProjectsPage_ = (props) => {
-  const canCreate = props.flags[FLAGS.CAN_CREATE_PROJECT];
+export const ProjectsPage = connectToFlags(FLAGS.CAN_CREATE_PROJECT)(({ flags, ...rest }) => {
   // Skip self-subject access review for projects since they use a special project request API.
   // `FLAGS.CAN_CREATE_PROJECT` determines if the user can create projects.
+  const canCreate = flags[FLAGS.CAN_CREATE_PROJECT];
+  const ListComponent = (listProps) => <ProjectList canCreate={canCreate} {...listProps} />;
```

Comment:
This might result in the component unmounting and remounting on each render. I think it's better to leave `ProjectList` a connected component as before.

Reply:
Okay, I figured since we were already getting the flag in the wrapping component, we could just pass it down.